modules/runtime/nuxeo-common/src/main/java/org/nuxeo/common/xmap/registry/MapRegistry.java

```diff
@@ -129,4 +129,12 @@ protected String computeId(Context ctx, XAnnotatedObject xObject, Element elemen
         return (T) contrib;
     }
 
+    protected void logError(String message, Throwable t, String extensionId, String id) {
+        log.error(message, t);
+    }
+
+    protected void logWarn(String message, String extensionId, String id) {
+        log.warn(message);
+    }
```

Comment:
Logging shouldn't exist in methods.

Also, this increases the complexity as we're now having two more interfaces to provide default methods + two more classes to link the abstract and the interface.

I would prefer to use a decorator pattern for this.

Reply:
I don't understand the "shouldn't exist" part. Please tell if you have another implementation in mind: i started defining a "RuntimeLogger" class and thought it was overkill after all.
I can remove the whole feature for now if you prefer: the point is to easily check that we have no implicit merges by default, but it's also to prepare for consistent exception management on classes instantiation (https://jira.nuxeo.com/browse/NXP-30092)

Reply:
I wanted to say that method referring to logging shouldn't exist in code.

I really think a decorator class would be more elegant.

After a Slack talk, we agree to postpone this.

Reply:
Commit moved to the wip branch for NXP-30092